pkg/config/config.go

```diff
@@ -145,3 +150,49 @@ func configFileWriter() (io.WriteCloser, error) {
 	}
 	return f, nil
 }
+
+const depthQueryParam = int32(5)
+
+type Client struct {
+	CloudClient       *cloudv6.APIClient
+	AuthClient        *sdkgoauth.APIClient
+	CertManagerClient *certmanager.APIClient
+	DbaasClient       *dbaas.APIClient
+}
+
+func NewClient(name, pwd, token, hostUrl string) (*Client, error) {
+	if token == "" && (name == "" || pwd == "") {
+		return nil, errors.New("username, password or token incorrect")
+	}
+	clientConfig := cloudv6.NewConfiguration(name, pwd, token, hostUrl)
+	clientConfig.UserAgent = fmt.Sprintf("%v_%v", viper.GetString(CLIHttpUserAgent), clientConfig.UserAgent)
+	// Set Depth Query Parameter globally
+	clientConfig.SetDepth(depthQueryParam)
+
+	authConfig := sdkgoauth.NewConfiguration(name, pwd, token, hostUrl)
+	authConfig.UserAgent = fmt.Sprintf("%v_%v", viper.GetString(CLIHttpUserAgent), authConfig.UserAgent)
+
+	certManagerConfig := certmanager.NewConfiguration(name, pwd, token, hostUrl)
+	certManagerConfig.UserAgent = fmt.Sprintf("%v_%v", viper.GetString(CLIHttpUserAgent), certManagerConfig.UserAgent)
+
+	dbaasConfig := dbaas.NewConfiguration(name, pwd, token, hostUrl)
+	dbaasConfig.UserAgent = fmt.Sprintf("%v_%v", viper.GetString(CLIHttpUserAgent), dbaasConfig.UserAgent)
+
+	return &Client{CloudClient: cloudv6.NewAPIClient(clientConfig), AuthClient: sdkgoauth.NewAPIClient(authConfig),
+		CertManagerClient: certmanager.NewAPIClient(certManagerConfig), DbaasClient: dbaas.NewAPIClient(dbaasConfig)}, nil
+}
+
+var once sync.Once
+var instance *Client
+
+func GetClient() (*Client, error) {
+	var err error
+	once.Do(func() {
+		err = Load()
+		instance, err = NewClient(viper.GetString(Username), viper.GetString(Password), viper.GetString(Token), GetServerUrl())
+	})
+	if err != nil {
+		return nil, err
+	}
+	return instance, nil
+}
```

Comment:
very nice! 

To make sure we're good in the future too, could you please add some test cases for this func? We need to test that empty credentials throws an understandable error, and we also need to check that a single Client is created even though GetClient is called however many times.